packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -48,19 +64,33 @@ def draw(self):
 
     def get_pixel_data(self):
         canvas = self.get_element('canvas')
-        ctx = canvas.getContext("2d")
-        canvas_data = ctx.getImageData(0, 0, int(ctx.width),
-                                       int(ctx.height)).data
-        canvas_data = np.asarray(canvas_data, dtype=np.uint8)
-        pixel_data = np.reshape(canvas_data, (int(ctx.height),
-                                int(ctx.width), 4))
-        return pixel_data
+        img_URL = canvas.toDataURL('image/png')[21:]
+        canvas_base64 = base64.b64decode(img_URL)
+        return _png.read_png_int(io.BytesIO(canvas_base64))
```

Comment:
Can you add a comment here about why it takes the long way round?  pixels -> encode png -> decode png -> pixels.  (Because not doing it results in different data than the reference image...)